include/clang/AST/Expr.h

```diff
@@ -4514,6 +4514,120 @@ class GenericSelectionExpr : public Expr {
   friend class ASTStmtReader;
 };
 
+/// \brief Represents a Checked C bounds expression.
+class BoundsExpr : public Expr {
+private:
+  SourceLocation StartLoc, RParenLoc;
+
+public:
+  BoundsExpr(StmtClass StmtClass, SourceLocation StartLoc, SourceLocation RParenLoc)
+    : Expr(StmtClass, QualType(),  VK_RValue, OK_Ordinary, false,
+           false, false, false), StartLoc(StartLoc), RParenLoc(RParenLoc) {
+  }
+
+  explicit BoundsExpr(StmtClass StmtClass, EmptyShell Empty) :
```

Comment:
This is used for deserialization of IR, where you need to create an IR node and then fill it in.

From: Reuben Olinsky [mailto:notifications@github.com]
Sent: Tuesday, June 28, 2016 5:06 PM
To: Microsoft/checkedc-clang <checkedc-clang@noreply.github.com>
Cc: David Tarditi <dtarditi@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Microsoft/checkedc-clang] Extend clang IR with bounds expressions and parse bounds expressions for parameters. (#8)


In include/clang/AST/Expr.h<https://github.com/Microsoft/checkedc-clang/pull/8#discussion_r68865737>:

> @@ -4514,6 +4514,120 @@ class GenericSelectionExpr : public Expr {

>    friend class ASTStmtReader;

>  };

>

> +/// \brief Represents a Checked C bounds expression.

> +class BoundsExpr : public Expr {

> +private:

> +  SourceLocation StartLoc, RParenLoc;

> +

> +public:

> +  BoundsExpr(StmtClass StmtClass, SourceLocation StartLoc, SourceLocation RParenLoc)

> +    : Expr(StmtClass, QualType(),  VK_RValue, OK_Ordinary, false,

> +           false, false, false), StartLoc(StartLoc), RParenLoc(RParenLoc) {

> +  }

> +

> +  explicit BoundsExpr(StmtClass StmtClass, EmptyShell Empty) :

Pardon my ignorance, but what does this do?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Microsoft/checkedc-clang/pull/8/files/340bbc6733cc1e11b2f0822b2bbc4f7e32fd87b9#r68865737>, or mute the thread<https://github.com/notifications/unsubscribe/ALSuCuX11xNDRRsbo41QkmOFSCMgzyS9ks5qQbcAgaJpZM4I_o4B>.